fireapp/lib/domain/repository/notification_fcm_token_repository.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:fireapp/data/client/fcm_token_client.dart';
+import 'package:fireapp/domain/models/notification/fcm_token_unregister.dart';
+import 'package:injectable/injectable.dart';
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'dart:io';
+
+import '../models/notification/fcm_token.dart';
+import 'authentication_repository.dart';
+
+@injectable
+class NotificationFCMTokenRepository {
+  final FCMTokenClient tokenClient;
+  final AuthenticationRepository _authenticationRepository;
+  NotificationFCMTokenRepository(
+      this.tokenClient, this._authenticationRepository);
+
+  Future<void> registerFCMToken(String token) async {
+    String deviceType = 'android';
+    var userId = await getUserId();
+    var curToken = FCMToken(fcmToken: token, deviceType: deviceType);
+    tokenClient.registerFCMToken(userId, curToken);
+  }
+
+  Future<void> unregisterToken(String token) async {
+    int userId = await getUserId();
+    await tokenClient.unregisterFCMToken(
+        userId, FCMTokenUnregister(fcmToken: token));
+  }
+
+  Future<int> getUserId() async {
```

Comment:
I'm not really happy with this being here, and if it is it definitely shouldn't be public. Add a helper function to AuthenticationRepository to get the userId (make the return nullable, like `getCurrentSession` is), then just inline the null check and exception.

Also please subclass Exception with a more specific one like `SignedOutException`. I don't really mind just using bare `Exception` in the ViewModel, since the only consumer is the UI. But in the repository, we should be returning more semantic exceptions since we may want to catch and handle them in some business logic layer.